### PR TITLE
COC-30 | Added optional initial modal to handle consent

### PR DESCRIPTION
**Pull request by MontiMarco92**

### CODE REVIEW

File: src/lib/Interface.js [hunk 1]
```diff
@@ -191,6 +191,26 @@ export default class Interface {
...
+        el('div.ccm__footer',
+          el('button.consent-decline', Language.getTranslation(window.CookieConsent.config, window.CookieConsent.config.language.current, 'initModalBtnDisagree')),
+
+          el('button.ccm__edit', Language.getTranslation(window.CookieConsent.config, window.CookieConsent.config.language.current, 'initModalBtnOptions')),
+          el('button.consent-give', Language.getTranslation(window.CookieConsent.config, window.CookieConsent.config.language.current, 'initModalBtnAgree')),
+        )
+      ), { 'aria-labelledby': 'ccm__content__title', 'aria-hidden': 'true' }
+    );
+  }
+
```
Comment (l-besenyei) [inline src/lib/Interface.js:213]: This creates a base dark overlay, and when opening the settings another overlay is added making it darker.
Let's make sure there is only one overlay.

File: src/lib/Configuration.js [hunk 2]
```diff
@@ -19,9 +19,10 @@ export default class Configuration {
       active: true,
       cookieExists: false,
       cookieVersion: 1,
+      modalLayoutActive: false,
```
Comment (l-besenyei) [inline src/lib/Configuration.js:22]: I would prefer a property called `mode`  with two options: 'bar', 'modal'

**Review by l-besenyei — CHANGES_REQUESTED** [review]

File: src/lib/Interface.js (hunk 1 above)
Comment (MontiMarco92) [reply src/lib/Interface.js:213]: Both modals are shown/hidden as needed, so the two overlays are not shown together causing the darkening effect

File: src/lib/Configuration.js (hunk 2 above)
Comment (MontiMarco92) [reply src/lib/Configuration.js]: done

**Review by l-besenyei — APPROVED** [review]

**Pull request merged**